nifi-commons/nifi-socket-utils/src/main/java/org/apache/nifi/io/socket/SocketUtils.java

```diff
@@ -185,4 +187,23 @@ public static void closeQuietly(final ServerSocket serverSocket) {
         }
     }
 
+    /**
+     * Returns {@code true} if this exception is due to a TLS problem (either directly or because of its cause, if present). Traverses the cause chain recursively.
+     *
+     * @param e the exception to evaluate
+     * @return true if the direct or indirect cause of this exception was TLS-related
+     */
+    public static boolean isTlsError(Throwable e) {
```

Comment:
This would make a good generic utility method for getting to the root cause of an issue. Could be done as a follow-on Jira though.